app/knight.py

```diff
@@ -0,0 +1,31 @@
+from typing import Dict, Any
+from app.armour import Armour
+from app.weapon import Weapon
+from app.potion import Potion
+
+
+class Knight:
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.name = data["name"]
+        self.hp = data["hp"]
+        self.power = data["power"]
+        self.armour = Armour(data.get("armour", Any))
+        self.weapon = Weapon(
+            data.get("weapon", {}).get("name", ""),
+            data.get("weapon", {}).get("power", 0)
+        )
+        self.potion = Potion(
+            data["potion"]["name"],
+            data["potion"]["effect"]
+        ) if data["potion"] else None
+        self.apply_potion()
+        self.calculate_stats()
+
+    def apply_potion(self) -> None:
+        if self.potion:
+            self.hp += self.potion.effect.get("hp", 0)
+            self.power += self.potion.effect.get("power", 0)
+            self.armour.protection += self.potion.effect.get("protection", 0)
```

Comment:
The apply_potion method is well implemented, but it would be better to add a check to ensure that the 'effect' key exists in the potion object before trying to access its properties. This will help avoid potential KeyError exceptions.